Source/Core/Core/HW/GCMemcard/GCIFile.cpp

```diff
@@ -87,16 +87,10 @@ int GCIFile::UsesBlock(u16 block_num)
 
 void GCIFile::DoState(PointerWrap& p)
 {
-  p.DoPOD<DEntry>(m_gci_header);
+  p.Do(m_gci_header);
   p.Do(m_dirty);
   p.Do(m_filename);
-  int num_blocks = (int)m_save_data.size();
-  p.Do(num_blocks);
-  m_save_data.resize(num_blocks);
-  for (auto itr = m_save_data.begin(); itr != m_save_data.end(); ++itr)
-  {
-    p.DoPOD<GCMBlock>(*itr);
-  }
+  p.Do(m_save_data);
```

Comment:
Note: this is mostly equivalent (as long as `num_blocks` is less than `INT_MAX`, and calling `Do` n times is the same as calling a larger `DoArray`, which should be the case for a trivially copyable type):

https://github.com/dolphin-emu/dolphin/blob/de5a98a3fa7c5df1d8627667f5bab192ecd1c0e9/Source/Core/Common/ChunkFile.h#L125-L129
https://github.com/dolphin-emu/dolphin/blob/de5a98a3fa7c5df1d8627667f5bab192ecd1c0e9/Source/Core/Common/ChunkFile.h#L319-L328
https://github.com/dolphin-emu/dolphin/blob/de5a98a3fa7c5df1d8627667f5bab192ecd1c0e9/Source/Core/Common/ChunkFile.h#L198-L202

Reply:
While I think you're right that this is equivalent, I think I'd feel safer if you bumped up the savestate version anyway -- #11125 is going to bump it too so it's no big deal to have a break here.

Reply:
Done. I also changed the similar manual `std::vector` handling in `GCMemcardDirectory.cpp`.